js/src/lib/components/InferenceWidget/shared/helpers.ts

```diff
@@ -3,30 +3,18 @@ import { randomItem, parseJSON } from "../../../utils/ViewUtils";
 import type { WidgetExample } from "./WidgetExample";
 import type { ModelLoadInfo, TableData } from "./types";
 
-const VALID_QUERY_PARAMS = [
-	// string vals
-	"text",
-	"context",
-	"question",
-	"query",
-	"candidateLabels",
-	// table vals
-	"multiClass",
-	"table",
-	// boolean vals
-	"structuredData",
-];
-export type QueryParam = typeof VALID_QUERY_PARAMS[number];
+type KeysOfUnion<T> = T extends any ? keyof T : never;
+export type QueryParam = KeysOfUnion<WidgetExample>;
 type QueryParamVal = string | null | boolean | (string | number)[][];
 export function getQueryParamVal(key: QueryParam): QueryParamVal {
 	const searchParams = new URL(window.location.href).searchParams;
 	const value = searchParams.get(key);
-	if (["text", "context", "question", "query", "candidateLabels"].includes(key)) {
+	if (["text", "context", "question", "query", "candidate_labels"].includes(key)) {
 		return value;
-	} else if (["table", "structuredData"].includes(key)) {
+	} else if (["table", "structured_data"].includes(key)) {
```

Comment:
same as the comment https://github.com/huggingface/hub-docs/pull/1023/commits/c89086b0ebea8a6968c7bc621134982270dbdbe6#r1376034236 but for `structuredData` vs `structured_data`

Reply:
https://github.com/huggingface/hub-docs/pull/1023/files/c89086b0ebea8a6968c7bc621134982270dbdbe6#r1376287272